desktop-exporter/app/routes/trace-view.tsx

```diff
@@ -20,62 +20,31 @@ export async function traceLoader({ params }: any) {
 export default function TraceView() {
   let traceData = useLoaderData() as TraceData;
   let traceDurationNs = getTraceDurationNs(traceData.spans);
-  let orderedSpans: SpanWithMetadata[] = [];
-  let spanTree: TreeItem[] = [];
+  let spanTree: TreeItem[] = arrayToTree(traceData.spans);
+  let orderedSpans = orderSpans(spanTree);
 
-  spanTree = arrayToTree(traceData.spans);
-
-  for (let root of spanTree) {
-    let stack = [
-      {
-        treeItem: root,
-        depth: 0,
-      },
-    ];
-
-    while (stack.length) {
-      let node = stack.pop();
-      if (!node) {
-        break;
-      }
-      let { treeItem, depth } = node;
-
-      orderedSpans.push({
-        spanID: treeItem.span.spanID,
-        spanData: treeItem.span.spanData,
-        metadata: { depth: node.depth },
-      });
+  let [selectedSpanID, setSelectedSpanID] = React.useState<string>(() => {
+    if (
+      !orderedSpans.length ||
+      (!orderedSpans[0].status && orderedSpans.length < 2)
+    ) {
+      return "";
+    }
 
-      treeItem.children
-        .sort((a, b) => {
-          let aStart = a.span.spanData
-            ? getNsFromString(a.span.spanData.startTime)
-            : 0;
-          let bStart = b.span.spanData
-            ? getNsFromString(b.span.spanData.startTime)
-            : 0;
-
-          if ((aStart = 0)) {
-            return bStart - aStart;
-          }
-          return aStart - bStart;
-        })
-        .forEach((child: TreeItem) =>
-          stack.push({
-            treeItem: child,
-            depth: depth + 1,
-          }),
-        );
+    if (!orderedSpans[0].status) {
+      return orderedSpans[1].metadata.spanID;
     }
-  }
 
-  let [selectedSpanID, setSelectedSpanID] = React.useState<string>(
-    orderedSpans.length ? orderedSpans[0].spanID : "",
-  );
+    return orderedSpans[0].metadata.spanID;
+  });
 
   // if we get a new trace because the route changed, reset the selected span
   React.useEffect(() => {
-    setSelectedSpanID(orderedSpans[0].spanID);
+    setSelectedSpanID(
+      orderedSpans[0].status
+        ? orderedSpans[0].metadata.spanID
+        : orderedSpans[1].metadata.spanID,
```

Comment:
Just to check, will we ever have a situation where both of these are missing? 🤔 I think no?

To be safer we might try something like:

```typescript
orderedSpans.find(span => span.status === SpanDataStatus.present).metadata.spanID
```

Except Typescript is going to complain that the `find` result could be `null` 🤔 

Maybe just leave it as-is